c++11/test/recorder_test.cpp

```diff
@@ -0,0 +1,45 @@
+#include <lightstep/tracer.h>
+#include "../src/buffered_recorder.h"
+#include "../src/lightstep_tracer_impl.h"
+#include "in_memory_transporter.h"
+#include "span_generator.h"
+
+#define CATCH_CONFIG_MAIN
+#include <lightstep/catch/catch.hpp>
+
+using namespace lightstep;
+using namespace opentracing;
+
+TEST_CASE("rpc_recorder") {
+  spdlog::logger logger{"lightstep", spdlog::sinks::stderr_sink_mt::instance()};
+  LightStepTracerOptions options;
+  options.reporting_period = std::chrono::milliseconds(2);
+  options.max_buffered_spans = 5;
+  auto in_memory_transporter = new InMemoryTransporter();
+  auto recorder = new BufferedRecorder{
+      logger, options, std::unique_ptr<Transporter>{in_memory_transporter}};
+  auto tracer = std::shared_ptr<opentracing::Tracer>{
+      new LightStepTracerImpl{std::unique_ptr<Recorder>{recorder}}};
+  CHECK(tracer);
+
+  SECTION(
+      "If spans are recorded at a rate significantly slower than "
+      "LightStepTracerOptions::reporting_period, then no spans get dropped.") {
+    SpanGenerator span_generator(*tracer, 2 * options.reporting_period);
+    span_generator.Run(std::chrono::milliseconds(250));
```

Comment:
I don't see how that example API would translate to this use case. The timing information (e.g. reporting period) is passed in to the Recorder class upon construction. Internally, the class manages a thread that periodically waits and runs. The point of this testing is to simulate scenarios that both record spans at a rate slower and faster than the reporting period can manage and to do so in a multithreaded context so that thread sanitizer might catch synchronization bugs. It's not for testing timeouts.

If you can think of a design that accomplishes that same coverage, doesn't sleep, doesn't add significant complexity, and is still close to what the actual code would be doing in a real deployment, I'm certainly open to considering it.